ForgeUnity/Assets/BeardedManStudios/Scripts/StandAlone/StandAloneObjectMapper.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+
+namespace BeardedManStudios.Forge.Networking.StandAlone
+{
+	public class StandAloneObjectMapper : ObjectMapper
```

Comment:
I'm wondering if it would be good to roll the mapping of these new types into the ObjectMapper itself instead. :thinking:

Reply:
I don't see many people needing the use for it as most people will just use unity headless servers which is why I put it in its own object mapper that inherits from the base

Reply:
The objectmapper not the unityobject mapper that extends the object mapper.

Reply:
Could it be that if you use `Float3` in RPCs, you will get errors whilst trying to send them from a Unity client?

```
BaseNetworkException: The type BeardedManStudios.Float3 is not allowed to be sent over the Network (yet)
BeardedManStudios.Forge.Networking.ObjectMapper.GetBytesArray (System.Object o, System.Type type) (at Assets/BeardedManStudios/Scripts/Networking/Forge/Networking/ObjectMapper.cs:455)
BeardedManStudios.Forge.Networking.Unity.UnityObjectMapper.GetBytesArray (System.Object o, System.Type type) (at Assets/BeardedManStudios/Scripts/UnityObjectMapper.cs:160)
BeardedManStudios.Forge.Networking.ObjectMapper.MapBytes (BeardedManStudios.BMSByte bytes, System.Object[] args) (at Assets/BeardedManStudios/Scripts/Networking/Forge/Networking/ObjectMapper.cs:217)
BeardedManStudios.Forge.Networking.NetworkObject.SendRpc (BeardedManStudios.Forge.Networking.NetworkingPlayer targetPlayer, System.Byte methodId, System.Boolean replacePrevious, System.Boolean reliable, BeardedManStudios.Forge.Networking.Receivers receivers, BeardedManStudios.Forge.Networking.NetworkingPlayer sender, System.Object[] args) (at Assets/BeardedManStudios/Scripts/Networking/Forge/Networking/Objects/NetworkObject.cs:1245)
BeardedManStudios.Forge.Networking.NetworkObject.SendRpc (System.Byte methodId, BeardedManStudios.Forge.Networking.Receivers receivers, System.Object[] args) (at Assets/BeardedManStudios/Scripts/Networking/Forge/Networking/Objects/NetworkObject.cs:1030)
MyNetworkTest.Update () (at Assets/Scripts/Network/MyNetworkTest.cs:96)
```

This seems to happen because `SendRpc` uses `ObjectMapper.Instance`, which is not an instance of `StandAloneObjectMapper`. Though I'm wondering how this _does_ work for `Vector3`, as those are only in the `UnityObjectMapper`, is the instance perhaps set to an instance of the child class somewhere?

Reply:
yup you found it, that is how it is in unity land.